### ISSUE TITLE
CHTTP replaces Content-Type header

### ISSUE THREAD

**Issue by Kotyarishka** [timschumi/gmod-chttp#30]
Hello! I was tinkering with discord webhooks, and got error about Content-Type header not being `application/json` when I clearly set propper header value.

After some investigation, I got this results, it is probably a bug or I'm missing something.

```lua
local request = {
        url = self._url,

        method = "post",
        
        headers = {
            ["Content-Type"] = "application/json; charset=utf-8",
            ["Custom-Header"] = "test value"
        },
        body = data,
        
        success = function(code, body, headers)
            print("Discord webhook sent successfully!")
        end,
        failed = function(err)
            print("Discord webhook failed to send!")
        end
 }

 CHTTP(request)
 ```
yielded this results:
![image](https://github.com/timschumi/gmod-chttp/assets/44447172/8dd0ce6a-0132-469c-bd6c-e859bdc143c6)

**Comment by timschumi** [comment]
If a `body` is provided, the value of the `type` parameter (which defaults to `text/plain; charset=utf-8`) takes precedence over the manually set `Content-Type`. As far as I am aware, this mirrors the behavior of the built-in `HTTP`.

**Comment by Kotyarishka** [comment]
Thanks for your reply. Tested with built-in `HTTP()` and there no such behavior.

As a workaround, I found that it is possible to pass `type` in the request table that will overwrite what `CHTTP()` is doing. Not sure if this is intended behavior, too, though.

**Comment by timschumi** [comment]
I just tested this just to make sure and this is _not_ how built-in HTTP behaves. HTTP gives higher precedence to the manually set `Content-Type` header. I have to work on some CI changes before that, but I'll most likely put out a bug fix tomorrow.

**Comment by timschumi** [comment]
Fixed in 220fa28ab1e51bfe3d075020cc748a1f0d9442da and released in [version 1.10.2](https://github.com/timschumi/gmod-chttp/releases/tag/v1.10.2).
